pkg/netutils/ips.go

```diff
@@ -4,51 +4,91 @@ import (
 	"fmt"
 	"net"
 	"strings"
-
-	"github.com/sirupsen/logrus"
 )
 
-// GetDefaultIPNet returns the default interface for the node, and the subnet mask for that node, using the same logic
-// as k0s in https://github.com/k0sproject/k0s/blob/v1.30.4%2Bk0s.0/internal/pkg/iface/iface.go#L61
-func GetDefaultIPNet() (*net.IPNet, error) {
-	ifs, err := net.Interfaces()
+// FirstValidAddress returns the first found non-local IPv4 address
+// that's not part of pod network for the given interface.
+// If an interface is not provided, the first found interface with a valid address is used.
+func FirstValidAddress(iface string) (string, error) {
+	ipnet, err := FirstValidIPNet(iface)
+	if err != nil {
+		return "", fmt.Errorf("get ipnet for interface %s: %w", iface, err)
+	}
+	if ipnet.IP.To4() == nil {
+		return "", fmt.Errorf("interface %s has no ipv4 addresses", iface)
+	}
+	return ipnet.IP.String(), nil
+}
+
+func FirstValidIPNet(iface string) (*net.IPNet, error) {
+	ifs, err := listValidInterfaces()
 	if err != nil {
-		return nil, fmt.Errorf("failed to list network interfaces: %w", err)
+		return nil, fmt.Errorf("list valid network interfaces: %w", err)
+	}
+	if len(ifs) == 0 {
+		return nil, fmt.Errorf("no valid network interfaces found on this machine")
+	}
+	if iface == "" {
+		return firstValidIPNet(ifs[0])
 	}
 	for _, i := range ifs {
-		if isPodnetworkInterface(i.Name) {
-			continue
+		if i.Name == iface {
+			return firstValidIPNet(i)
 		}
-		addresses, err := i.Addrs()
-		if err != nil {
-			logrus.Debugf("failed to get addresses for interface %s: %s", i.Name, err.Error())
+	}
+	return nil, fmt.Errorf("interface %s not found or is not valid", iface)
```

Comment:
```suggestion
	ifaceNames := []string{}
	for _, i := range ifs {
		ifaceNames = append(ifaceNames, i.Name)
	}
	return nil, fmt.Errorf("interface %s not found or is not valid. The following interfaces were detected %v", iface, ifaceNames)
```

this should probably be integrated with the previous loop but github wouldn't let me suggest that